test/e2e/alertmanager_test.go

```diff
@@ -291,3 +285,186 @@ receivers:
 		t.Fatal(err)
 	}
 }
+
+func TestAlertmanagerZeroDowntimeRollingDeployment(t *testing.T) {
+	t.Parallel()
+
+	ctx := framework.NewTestCtx(t)
+	defer ctx.Cleanup(t)
+	ns := ctx.CreateNamespace(t, framework.KubeClient)
+
+	whReplicas := int32(1)
+	whdpl := &v1beta1.Deployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "alertmanager-webhook",
+		},
+		Spec: v1beta1.DeploymentSpec{
+			Replicas: &whReplicas,
+			Template: v1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: map[string]string{
+						"app": "alertmanager-webhook",
+					},
+				},
+				Spec: v1.PodSpec{
+					Containers: []v1.Container{
+						{
+							Name:  "webhook-server",
+							Image: "quay.io/brancz/prometheus-alertmanager-test-webhook",
+							Ports: []v1.ContainerPort{
+								{
+									Name:          "web",
+									ContainerPort: 5001,
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+	whsvc := &v1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "alertmanager-webhook",
+		},
+		Spec: v1.ServiceSpec{
+			Type: v1.ServiceTypeClusterIP,
+			Ports: []v1.ServicePort{
+				v1.ServicePort{
+					Name:       "web",
+					Port:       5001,
+					TargetPort: intstr.FromString("web"),
+				},
+			},
+			Selector: map[string]string{
+				"app": "alertmanager-webhook",
+			},
+		},
+	}
+	if _, err := framework.KubeClient.CoreV1().Services(ns).Create(whsvc); err != nil {
+		t.Fatal(err)
+	}
+	if _, err := framework.KubeClient.ExtensionsV1beta1().Deployments(ns).Create(whdpl); err != nil {
```

Comment:
Same here, can we use `CreateDeployment`. It might not add a lot of convenience at the moment, but maybe we want to add more *clean up* functionality in the long run, which would be easier when we use these functions everywhere. What do you think?